### PR TITLE
Add opentracing spans to calls to external cache

### PR DESCRIPTION
**Pull request by erikjohnston**
I don't think redis is causing any issues, but we should be adding spans whenever we talk to an external service.

### CODE REVIEW

**Comment by erikjohnston** [comment]
(CI is broken on develop)

File: synapse/replication/tcp/external_cache.py [hunk 1]
```diff
@@ -94,7 +94,10 @@ async def set(self, cache_name: str, key: str, value: Any, expiry_ms: int) -> No
 
         logger.debug("Caching %s %s: %r", cache_name, key, encoded_value)
 
-        with opentracing.start_active_span("ExternalCache.set"):
+        with opentracing.start_active_span(
+            "ExternalCache.set",
+            tags={opentracing.SynapseTags: cache_name},
```
Comment (clokep) [inline synapse/replication/tcp/external_cache.py:99]: I assume this should be?

```suggestion
            tags={opentracing.SynapseTags.CACHE_NAME: cache_name},
```

File: synapse/replication/tcp/external_cache.py [hunk 2]
```diff
@@ -110,7 +113,10 @@ async def get(self, cache_name: str, key: str) -> Optional[Any]:
         if self._redis_connection is None:
             return None
 
-        with opentracing.start_active_span("ExternalCache.get"):
+        with opentracing.start_active_span(
+            "ExternalCache.get",
+            tags={opentracing.SynapseTags: cache_name},
```
Comment (clokep) [inline synapse/replication/tcp/external_cache.py:118]:
```suggestion
            tags={opentracing.SynapseTags.CACHE_NAME: cache_name},
```

File: synapse/replication/tcp/external_cache.py (hunk 1 above)
Comment (erikjohnston) [reply synapse/replication/tcp/external_cache.py:99]: :facepalm:

**Review by clokep — APPROVED** [review]

**Pull request merged**